lib/web_ui/lib/src/engine/skwasm/skwasm_impl/paragraph.dart

```diff
@@ -940,27 +940,31 @@ class SkwasmParagraphBuilder extends SkwasmObjectWrapper<RawParagraphBuilder> im
     // just create both up front here.
     final Pointer<Uint32> outSize = scope.allocUint32Array(1);
     final Pointer<Uint8> utf8Data = paragraphBuilderGetUtf8Text(handle, outSize);
+
+    final String text;
+    final JSString jsText;
     if (utf8Data == nullptr) {
-      return;
+      text = '';
+      jsText = ''.toJS;
+    } else {
+      // TODO(jacksongardner): We could make a subclass of `List<int>` here to
+      // avoid this copy.
```

Comment:
I was going to suggest making using `Iterable<int>.generate(...)` instead, but then realized that `utf8.decode(...)` takes a `List<int>`. I wonder why.

Reply:
Typically anything but a raw `List` would be many times slower. I'm not surprised `utf8` prefers lists. Honestly, this is fine. I wouldn't even leave a TODO.

Reply:
That's probably true for dart2js since raw `List` are basically JS arrays under the hood (IIUC). But is it the same for wasm?

Reply:
It's not the same for wasm, `List` is definitely polymorphic. A while back I actually did try making a subclass of `List<int>` that reads from an ffi pointer, and it actually was slightly slower. I think I'll just remove this TODO.